bob-common/src/node.rs

```diff
@@ -78,6 +91,17 @@ impl Node {
         self.conn.read().await.clone()
     }
 
+    pub fn reset_error_count(&self) {
+        self.err_count.store(0, Ordering::Relaxed);
+    }
+
+    pub async fn increase_error_and_clear_conn_if_needed(&self) {
+        let count = self.err_count.fetch_add(1, Ordering::Relaxed);
+        if count >= self.max_sequential_errors {
+            self.clear_connection().await;
```

Comment:
This is incorrect because another thread may reset the counter. You must acquire the `write` lock and check the value, and only if it is still greater than `max_sequential_errors`, clear the connection.